src/Avalonia.FuncUI/Types.fs

```diff
@@ -75,11 +75,13 @@ module Types =
                 match other with
                 | :? Subscription as other -> 
                     this.name = other.name &&
-                    this.funcType = other.funcType
+                    this.funcType = other.funcType &&
+                    not this.funcCapturesState &&
```

Comment:
not sure if this is the right thing here.

In theory if a function that captures state should be of a different type than a function that doesn't (Because the compiler will generate a different class as far as I know). 

With this equals implementation two subscriptions that are basically equal will return false.

```fsharp
let sub = {
    Subscription.name = "sub"
    subscribe = fun _ -> new CancellationTokenSource()
    func = dlgt
    funcCapturesState = true
    funcType = typeof<int>
}

sub.Equals(sub) // this returns false
```

> x.Equals(x) returns true.
[microsoft docs](https://docs.microsoft.com/en-us/previous-versions/ms173147(v=vs.90)?redirectedfrom=MSDN)

Reply:
Sorry that I haven't merged your PR and also did not came up with a fix myself. I'm currently rethinking the whole subscription model and am trying to come up with a better implementation that doesn't require to patch basically all subscriptions on every view update.

There is another issue with subscriptions that leads to a loop of (patch -> render -> patch -> ..) because presubscription of a property invokes the subscription once.

Reply:
I think here https://github.com/Thecentury/Avalonia.FuncUI/commit/07df113485870ea51a7006093040393ddff1d741 in two subsequent calls of the view function `funcType` was the same, otherwise there would be no problem.

Reply:
>  I'm currently rethinking the whole subscription model

Nice to hear that :)

My fix is indeed a shallow one.